### PR TITLE
Weld used in a fat jar

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

**Comment by weld-tester** [comment]
Triggering build using a merge of a68e8ab2fa3b5deff854481bb49e5ceb49433298 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 859 is now running using a merge of a68e8ab2fa3b5deff854481bb49e5ceb49433298 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/859

**Comment by weld-tester** [comment]
Build 859 outcome was SUCCESS using a merge of a68e8ab2fa3b5deff854481bb49e5ceb49433298 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/859
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/859

**Comment by weld-tester** [comment]
Build 859 outcome was SUCCESS using a merge of a68e8ab2fa3b5deff854481bb49e5ceb49433298 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/859
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/859

**Comment by weld-tester** [comment]
Build 859 outcome was SUCCESS using a merge of a68e8ab2fa3b5deff854481bb49e5ceb49433298 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/859
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/859

**Comment by weld-tester** [comment]
Build 859 outcome was SUCCESS using a merge of a68e8ab2fa3b5deff854481bb49e5ceb49433298 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/859
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/859

**Comment by weld-tester** [comment]
Build 859 outcome was SUCCESS using a merge of a68e8ab2fa3b5deff854481bb49e5ceb49433298 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/859
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/859

**Comment by weld-tester** [comment]
Build 859 outcome was SUCCESS using a merge of a68e8ab2fa3b5deff854481bb49e5ceb49433298 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/859
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/859

File: impl/src/main/java/org/jboss/weld/util/reflection/Formats.java [hunk 1]
```diff
@@ -479,12 +483,38 @@ public static String formatAnnotations(Annotation[] annotations) {
...
-            throw new IllegalArgumentException("Package can not be null");
-        } else {
-            return version(pkg.getSpecificationVersion(), pkg.getImplementationVersion());
+    public static String version(Class<?> weldClass) {
+
+        String version = null;
+        String timestamp = null;
+
+        // First try to get weld-build.properties file
+        InputStream in = weldClass.getResourceAsStream(BUILD_PROPERTIES_FILE);
```
Comment (jharting) [inline impl/src/main/java/org/jboss/weld/util/reflection/Formats.java]: This could probably be simplified with try-with-resources

Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/util/reflection/Formats.java]: I was not sure about null-handling semantics of the try-with-resources statement (i.e. what happens if the resource is `null`). But it seems it should be save: https://blogs.oracle.com/darcy/entry/project_coin_null_try_with. I will update the pull.

**Comment by weld-tester** [comment]
Triggering build using a merge of b5007c5ae2a7f83f3a40a25bd9ea84813c22fc4e on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 866 is now running using a merge of b5007c5ae2a7f83f3a40a25bd9ea84813c22fc4e on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/866

**Comment by weld-tester** [comment]
Build 866 outcome was SUCCESS using a merge of b5007c5ae2a7f83f3a40a25bd9ea84813c22fc4e on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/866
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/866

**Comment by jharting** [comment]
Please send a separate PR for 2.3

**Comment by mkouba** [comment]
https://github.com/weld/core/pull/1036

**Comment by weld-tester** [comment]
Build 866 outcome was SUCCESS using a merge of b5007c5ae2a7f83f3a40a25bd9ea84813c22fc4e on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/866
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/866

**Pull request closed without merging**